modules/i3s/src/i3s-attribute-loader.js

```diff
@@ -25,3 +26,61 @@ async function parse(data, options) {
   data = parseI3STileAttribute(data, options);
   return data;
 }
+
+export function getAttributesFromTileByFeatureId(tile, featureId) {
+  if (featureId < 0 || !tile || !tile.header) {
+    return null;
+  }
+
+  if (!tile.header.userData.layerFeaturesAttributes) {
+    return null;
+  }
+
+  const {attributeStorageInfo} = tile.tileset.tileset;
+  const {layerFeaturesAttributes} = tile.header.userData;
+  const calculatedFeatureIndex = getFeatureAttributeIndex(layerFeaturesAttributes, featureId);
+
+  if (calculatedFeatureIndex < 0) {
+    return null;
+  }
+
+  return prepareFeatureAttributes(
+    attributeStorageInfo,
+    layerFeaturesAttributes,
+    calculatedFeatureIndex
+  );
+}
+
+function getFeatureAttributeIndex(layerFeaturesAttributes, featureId) {
+  let featureIndex = -1;
+
+  for (let index = 0; index < layerFeaturesAttributes.length; index++) {
```

Comment:
When we load the tile, perhaps we could just create a map `attributesByObjectId` in the generated tile so the application can do lookup without calling a function?

Reply:
Will do it in further PR